codegen/src/lib.rs

```diff
@@ -6,13 +6,65 @@ mod shuttle_main;
 use proc_macro::TokenStream;
 use proc_macro_error::proc_macro_error;
 
+/// Setups up the `shuttle_runtime` to be executed. Allows
+/// for user to setup a runtime without needing to explicitly
+/// create one.
+///
+/// # Example
+///
+/// ```
+/// #[shuttle_runtime::main]
+/// async fn entry_point() {
+///     todo!();
+/// }
+/// ```
 #[cfg(feature = "frameworks")]
 #[proc_macro_error]
 #[proc_macro_attribute]
 pub fn main(attr: TokenStream, item: TokenStream) -> TokenStream {
     shuttle_main::r#impl(attr, item)
 }
 
+/// Setups up a `shuttle_app` to be generated. Allows
+/// for user to setup a `next` application.
```

Comment:
Perhaps something like this here? And perhaps with links to axum router and hyper server?
```rust
/// Generates a wasm32-wasi module containing an Axum router with your endpoints, which is passed as a 
/// hyper::service::Service to a hyper::Server. 
```